src/PaymentGateways/Gateways/Stripe/CreditCardGateway.php

```diff
@@ -0,0 +1,144 @@
+            'give_stripe_create_intent_args',
+            [
+                'amount' => Money::of($paymentData->price, $paymentData->currency)->getMinorAmount(),
+                'currency' => $paymentData->currency,
+                'payment_method_types' => ['card'],
+                'statement_descriptor' => give_stripe_get_statement_descriptor(),
+                'description' => '', //give_payment_gateway_donation_summary( $donationData->toArray() ),
+                'metadata' => give_stripe_prepare_metadata($paymentData->donationId, $donationData->toArray()),
+                'customer' => $giveStripeCustomer->get_id(),
+                'payment_method' => $paymentMethodId,
+                'confirm' => true,
+                'return_url' => $paymentData->redirectUrl,
+            ]
+        );
+
+        // Send Stripe Receipt emails when enabled.
+        if ( give_is_setting_enabled( give_get_option( 'stripe_receipt_emails' ) ) ) {
+            $intent_args['receipt_email'] = $paymentData->donorInfo->email;
+        }
+
+        $intent = give( LegacyStripePaymentIntent::class )->create( $intent_args );
+
+        give_insert_payment_note( $paymentData->donationId, 'Stripe Charge/Payment Intent ID: ' . $intent->id );
+        give_insert_payment_note( $paymentData->donationId, 'Stripe Payment Intent Client Secret: ' . $intent->client_secret );
+        give_update_meta( $paymentData->donationId, '_give_stripe_payment_intent_client_secret', $intent->client_secret );
+
+        switch( $intent->status() )  {
+            case 'requires_action':
+                give_insert_payment_note( $paymentData->donationId, 'Stripe requires additional action to be fulfilled.' );
+                give_update_meta( $paymentData->donationId, '_give_stripe_payment_intent_require_action_url', $intent->nextActionRedirectUrl() );
+                return new RedirectOffsite( $intent->nextActionRedirectUrl() );
+            case 'succeeded':
+                return new PaymentProcessing( $intent->id() );
+            default:
+                throw new PaymentIntentException( sprintf( __( 'Unhandled payment intent status: %s', 'give' ), $intent->status() ) );
+        }
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public static function id()
+    {
+        return 'stripe';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getId()
+    {
+        return self::id();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getName()
+    {
+        return __('Stripe - Credit Card', 'give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getPaymentMethodLabel()
+    {
+        return __('Stripe - Credit Card', 'give');
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getLegacyFormFieldMarkup($formId)
+    {
+        if (FormUtils::isLegacyForm($formId)) {
+            return false;
+        }
+
+        // @TODO Migrate field markup from legacy gateway implementation.
```

Comment:
@kjohnson what's our plan here?  Is there any reason we can't move this over in this PR?

Reply:
@jonwaldstein I was considering doing that in a separate PR, moving a logical piece at a time.

Reply:
I would encourage moving that over now if you're doing one PR for each Stripe gateway variant so that it covers the majority of the gateway.  All depends on your workflow of course.

Reply:
Resolved by 9650161.